### DISCUSSION TITLE
How to handle desired hydration mismatches?

### DISCUSSION THREAD

**Discussion by ronenteva** [nuxt/nuxt#28095] [Questions]
When using a CDN cache, sometimes hydration mismatches are desirable, what is the right way of handling them?
 
Say I have the following component:
```
<script setup>
  const {date} = defineProps(['date']);
  const timeAgo = useTimeAgo(date);
</script>

<template>
  <time :datetime="date">{{ timeAgo }}</time>
</template>
```

The rendered page is **cached** on the CDN, which means a user on a different time zone than the server will get a hydration error:
```
[Vue warn]: Hydration text content mismatch on <time datetime="2024-07-09T23:31:06.807Z">7 hours ago</time>
 
  - rendered on server: 7 hours ago
  - expected on client: 8 hours ago
```

* Simply ignoring the hydration mismatch errors doesn't seem right.
* Wrapping the component in `<ClientOnly>` has SEO disadvantages.
* Using `onMounted` should work, but it feels hacky when used so often.

Any other options I should consider?

**Comment by geforseN** [comment]
Vue 3.5 introduced [data-allow-mismatch](https://blog.vuejs.org/posts/vue-3-5#data-allow-mismatch).
```html
<time :datetime="date" data-allow-mismatch>{{ timeAgo }}</time>
```
